Java/jni/native_glue.cc

```diff
@@ -84,6 +89,68 @@ namespace litecore {
                 free((void *) _slice.buf);        // detached
         }
 
+        void jstringSlice::fixBrokenModifiedUTF8(const char *input){
+            // https://github.com/android-ndk/ndk/issues/283
+            size_t len = strlen(input);
+            size_t newLen = len;
+            queue<uint8_t*> modified;
+            const auto unsignedInput = (const uint8_t *)input;
+
+            // Need to figure out the actual length since each
+            // 4-bytes of real UTF-8 is 6 bytes of modified UTF-8
+            // but convert the bytes while we are at it
+            for(size_t i = 0; i < len; i++) {
+                if(unsignedInput[i] >= 0xF0) {
+                    // 0xF0 and above marks 4-byte sequences
+                    newLen += 2;
+                    int unicodePoint = (unsignedInput[i]-240<<18) +
+                            (unsignedInput[i+1]-128<<12) +
+                            (unsignedInput[i+2]-128<<6) +
+                            (unsignedInput[i+3]-128) - 0x10000;
+                    uint16_t surrogates[2] { (uint16_t)0xD800 | (uint16_t)(unicodePoint >> 10),
+                                             (uint16_t)0xDC00 | (uint16_t)(unicodePoint & 0x3FF) };
+
+                    modified.emplace({
```

Comment:
I think there's a use-after-free bug here. This call implicitly creates a stack-based C array of uint8_t, then adds its address to `modified`. But the C array is invalidated at the end of this `for` loop iteration, so `modified` ends up being full of pointers to garbage.

Reply:
Emplacement should do an std::move I thought?

Reply:
Sure, but it's moving a _pointer_. If you wanted to copy the bytes, you'd need to use a `queue<string>` or something like that.

Reply:
Oh that's correct yeah, newb mistake on my part.